python/python/glide/redis_client.py

```diff
@@ -243,6 +255,80 @@ async def _execute_script(
         set_protobuf_route(request, route)
         return await self._write_request_await_response(request)
 
+    @dataclass
+    class PubSubMsg:
+        message: str
+        channel: str
+        pattern: Optional[str]
+
+    async def get_pubsub_message(self) -> PubSubMsg:
+        if self._is_closed:
+            raise ClosingError(
+                "Unable to execute requests; the client is closed. Please create a new client."
+            )
+        # locking might not be required
+        response_future: asyncio.Future = asyncio.Future()
+        try:
+            self._pubsub_lock.acquire()
+            self._pubsub_futures.append(response_future)
+            self._push_pubsub_messages_safe()
+        finally:
+            self._pubsub_lock.release()
+        return await response_future
+
+    def _cancel_pubsub_futures_with_exception_safe(self, exception: ConnectionError):
+        while len(self._pubsub_futures):
+            next_future = self._pubsub_futures.pop(0)
+            if not next_future.cancelled():
+                next_future.set_exception(exception)
+
+    def _push_pubsub_messages_safe(self):
+        while len(self._pending_push_notifications) and len(self._pubsub_futures):
+            next_push_notification = self._pending_push_notifications.pop(0)
+            next_push_notification: Dict = value_from_pointer(
+                next_push_notification.resp_pointer
+            )
+            message_kind = next_push_notification["kind"]
+            if message_kind == "Disconnect":
+                # cancel all futures since we dont know how many (if any) messages wont arrive
+                # TODO: consider cancelling a single future
+                self._cancel_pubsub_futures_with_exception_safe(
+                    ConnectionError(
+                        "Warning, transport disconnect occured, messages might be lost"
+                    )
+                )
+            elif (
+                message_kind == "Message"
```

Comment:
Please don't use magic strings, create constants for that

Reply:
generally , you are right, but in this case these string are not used anywere else and defined only once